lib/std/tempfiles.nim

```diff
@@ -82,13 +82,21 @@ template randomPathName(length: Natural): string =
     res[i] = state.sample(letters)
   res
 
+proc genTempPath*(prefix, suffix: string, dir = ""): string =
+  ## Generates a path name in `dir`.
+  ##
+  ## If `dir` is empty, (`getTempDir <os.html#getTempDir>`_) will be used.
+  ## The path begins with `prefix` and ends with `suffix`.
+  var dir = dir
+  if dir.len == 0:
+    dir = getTempDir()
+  result = dir / (prefix & randomPathName(nimTempPathLength) & suffix)
+
 proc createTempFile*(prefix, suffix: string, dir = ""): tuple[fd: File, path: string] =
```

Comment:
You need to clean up the body of `createTempFile`.

remove

```
  var dir = dir
  if dir.len == 0:
    dir = getTempDir()
```

Reply:
i've now refactored via getTempDirImpl which simplifies code, but i can't remove this otherwise it keeps getting recomputed in the loop